state/machine_ports_ops.go

```diff
@@ -15,11 +16,12 @@ var (
 )
 
 type openClosePortsOperation struct {
-	p               *Ports
-	openPortRanges  []PortRange
-	closePortRanges []PortRange
+	p               *machineSubnetPorts
+	unitName        string
+	openPortRanges  []network.PortRange
+	closePortRanges []network.PortRange
 
-	updatedPortList []PortRange
+	updatedPortList []portRangeDoc
 }
 
 // Build implements ModelOperation.
```

Comment:
This is verging on too large. I'd consider if it's worth breaking out the open/close processing loops.

Reply:
Extracted into separate methods in https://github.com/juju/juju/pull/11842/commits/2350c2c603d1296d27f93cd1a6ad183e62ef15bc